src/Aspire.Hosting/ApplicationModel/HttpCommandOptions.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Optional configuration for resource HTTP commands added with <see cref="ResourceBuilderExtensions.WithHttpCommand{TResource}(IResourceBuilder{TResource}, string, string, string?, HttpCommandOptions?)"/>."/>
+/// </summary>
+public class HttpCommandOptions : CommandOptions
+{
+    internal static new HttpCommandOptions Default { get; } = new();
+
+    /// <summary>
+    /// Gets or sets a callback that selects the HTTP endpoint to send the request to when the command is invoked.
+    /// </summary>
+    public Func<EndpointReference>? EndpointSelector { get; set; }
+
+    /// <summary>
+    /// Gets or sets the HTTP method to use when sending the request.
+    /// </summary>
+    public HttpMethod? Method { get; set; }
+
+    /// <summary>
+    /// Gets or sets the name of the HTTP client to use when creating it via <see cref="IHttpClientFactory.CreateClient(string)"/>.
+    /// </summary>
+    public string? HttpClientName { get; set; }
+
+    /// <summary>
+    /// Gets or sets a callback to be invoked to configure the request before it is sent.
+    /// </summary>
+    public Func<HttpCommandRequestContext, Task>? PrepareRequest { get; set; }
+
+    /// <summary>
+    /// Gets or sets a callback to be invoked after the response is received to determine the result of the command invocation.
+    /// </summary>
+    public Func<HttpCommandResultContext, Task<ExecuteCommandResult>>? GetCommandResult { get; set; }
+
+    /// <summary>
+    /// Gets or sets the name of command. The name uniquely identifies the command.
+    /// </summary>
+    public string? CommandName { get; set; }
```

Comment:
WithCommand already has a name. How does this interact with it? Also, why is command name specific to HttpCommandOptions? Shouldn't it be on the base CommandOptions?

Reply:
`CommandOptions` is for things that are optional as it's replacing all the optional parameters. You still need to pass a name to `WithCommand`. In the case of `WithHttpCommand`, a name is not required as a default will be generated for you, so it's optional on `HttpCommandOptions`.

Reply:
It feels weird that there are different ways for doing it for other commands and HTTP commands.

Maybe the name parameter is optional on WithHttpCommand. If you want to specify it then provide a value for the optional parameter.

Reply:
Sounds good.

Reply:
@JamesNK updated.